pyquil/api/_engagement_manager.py

```diff
@@ -102,6 +104,7 @@ def get_engagement(
                     except QCSHTTPStatusError as e:
                         if response.status_code == 503:
                             raise QPUUnavailableError(retry_after=response.headers.get("Retry-After")) from e
+                        raise e
             return self._cached_engagements[key]
```

Comment:
Note that safe dictionary access via `.get` is already ensured for the lock session here:

https://github.com/rigetti/pyquil/blob/717a12c110c71a8b9d701047f06f038a9dad6008/pyquil/api/_engagement_manager.py#L94

The addition of `raise e` will ensure that any error in the `try` block will still throw, so the `get_engagement` function only continues if the cache is populated.

Reply:
However, if we want to make sure that this access is guarded, we could do something similar to the following:

```suggestion
            engagement = self_cached_engagements.get(key)
            if not self._engagement_valid(engagement):
                // TODO: create more detailed error in this case
                raise Exception(f"expected valid engagement in cache, but got: {engagement}")
            return engagement
```

Reply:
> The addition of raise e will ensure that any error in the try block will still throw

Ah, yeah, this is fine without the extra check - with your addition in this PR, that path is safe again. 

Note, for code ⛳  purposes, `raise` is equivalent to `raise e`.